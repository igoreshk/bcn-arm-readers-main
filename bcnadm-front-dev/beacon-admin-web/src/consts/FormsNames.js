export const USER_DIALOG_FORM = 'UserForm';
export const VISITOR_DIALOG_FORM = 'VisitorForm';
export const BUILDING_DIALOG_FORM = 'BuildingForm';

/* Form field names */
export const LEVEL_FIELD = 'levels';
