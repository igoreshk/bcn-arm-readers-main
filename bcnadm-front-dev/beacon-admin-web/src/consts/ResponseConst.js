export const OK = 200;
export const FORBIDDEN = 403;
