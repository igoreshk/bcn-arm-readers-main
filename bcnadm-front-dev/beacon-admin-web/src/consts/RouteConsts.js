export const DEFAULT_INITIAL_PAGE = '/buildings';
export const ID = '/:id';
export const DELETE = '/delete';
export const EDIT = '/edit';
export const NEW = '/new';
export const NEW_BUILDING = '/new/building';
export const BUILDINGS_LIST = '/buildings';
export const BUILDINGS = '/buildings';
export const BUILDING_NEW = BUILDINGS_LIST + NEW;
export const BUILDING_DIALOG = BUILDINGS_LIST + ID;
export const BUILDING_DELETE_DIALOG = BUILDINGS_LIST + ID + DELETE;
export const USERS_LIST = '/users';
export const USERS_DELETE_DIALOG = USERS_LIST + ID + DELETE;
export const CURRENT = '/current';
export const CREATE_USER = USERS_LIST + NEW;
export const USER_DIALOG = USERS_LIST + ID;
export const VISITORS_LIST = '/visitors';
export const CREATE_VISITOR = VISITORS_LIST + NEW;
export const VISITOR_DIALOG = VISITORS_LIST + ID;
export const VISITORS_DELETE_DIALOG = VISITORS_LIST + ID + DELETE;
export const WELCOME_PAGE = '/';
export const LEVEL_ID = '/:level';
export const BUILDING_ID = '/:building';
export const BEACON_ID = '/:beacon';
export const READER_ID = '/:reader';
export const EDGE_ID = '/:edge';
export const AREA_ID = '/:area';
export const AREAS = '/areas';
export const READERS = '/readers';
export const ROUTING = '/routing';
export const BEACONS = '/beacons';
export const SCALING = '/scaling';
export const MONITORING = '/monitoring';
export const DRAWING = '/drawing';
export const MAP = '/map';
export const LEVELS = '/levels';
export const LAYER = '/:layer';
export const EDIT_MODE = '/:edit?';

export const INITIAL_PAGE = {
  'admin': USERS_LIST,
  'operator': BUILDINGS_LIST
};
export const EMAIL = '/email';
export const LOGIN = '/login';
export const PASSWORD = '/password';
export const USER = '/user';
export const MESSAGES = '/messages';
export const PROFILE = '/profile';
export const USER_PROFILE = USER + PROFILE;
export const USER_MESSAGES = USER + MESSAGES;
export const WATCHERS_LIST = '/watchers';
export const CREATE_WATCHER = WATCHERS_LIST + NEW;
export const WATCHER_DIALOG = WATCHERS_LIST + ID;
export const WATCHERS_DELETE_DIALOG = WATCHERS_LIST + ID + DELETE;
