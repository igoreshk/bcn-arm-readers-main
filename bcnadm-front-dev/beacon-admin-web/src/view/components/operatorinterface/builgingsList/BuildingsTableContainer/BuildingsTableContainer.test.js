import { BUILDINGS_LIST } from 'src/consts/RouteConsts';
import React from 'react';
import { shallow } from 'enzyme';
import { BuildingsTableContainer } from './index';

const props = {
  translate: jest.fn(),
  showLoadingScreen: jest.fn(),
  hideLoadingScreen: jest.fn(),
  history: {
    push: jest.fn()
  },
  updated: false
};

const filteredBuildings = [{ entityId: 1 }];

const wrapper = shallow(<BuildingsTableContainer {...props} />);
const loadData = jest.spyOn(BuildingsTableContainer.prototype, 'loadData');
const showLoadingScreen = jest.spyOn(wrapper.instance().props, 'showLoadingScreen');

describe('BuildingsTableContainer', () => {
  it('Should check that props are passed properly', () => expect(wrapper.instance().props).toEqual(props));

  it('Should call showLoadingScreen', () => {
    wrapper.setProps({ updated: true });
    expect(showLoadingScreen).toHaveBeenCalledTimes(2);
  });

  it('Checks if history is called with correct arguments', () => {
    const event = {
      stopPropagation: jest.fn()
    };
    wrapper.instance().onEditClick(filteredBuildings[0])(event);

    expect(wrapper.instance().props.history.push).toHaveBeenCalledWith(
      `${BUILDINGS_LIST}/${filteredBuildings[0].entityId}`
    );
  });

  it('Should call loadData', () => expect(loadData).toHaveBeenCalled());

  it('Should match snapshot', () => expect(wrapper).toMatchSnapshot());
});
