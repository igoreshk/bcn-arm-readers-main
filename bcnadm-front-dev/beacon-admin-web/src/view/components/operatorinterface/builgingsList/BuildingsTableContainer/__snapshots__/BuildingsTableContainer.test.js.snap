// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildingsTableContainer Should match snapshot 1`] = `
<Fragment>
  <WithSearch(BuildingsTable)
    keys={
      Array [
        "name",
        "address",
      ]
    }
    onEditClick={[Function]}
    onRemoveClick={[Function]}
    showMap={[Function]}
    source={Array []}
    translate={
      [MockFunction] {
        "calls": Array [
          Array [
            "building.messages.noLevels",
          ],
          Array [
            "building.addLevel",
          ],
          Array [
            "building.messages.noLevels",
          ],
          Array [
            "building.addLevel",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
  />
  <WithStyles(ForwardRef(Snackbar))
    action={
      <React.Fragment>
        <WithStyles(ForwardRef(Button))
          color="secondary"
          onClick={[Function]}
          size="small"
        />
        <WithStyles(ForwardRef(IconButton))
          aria-label="close"
          color="inherit"
          onClick={[Function]}
          size="small"
        >
          <Memo
            fontSize="small"
          />
        </WithStyles(ForwardRef(IconButton))>
      </React.Fragment>
    }
    anchorOrigin={
      Object {
        "horizontal": "left",
        "vertical": "bottom",
      }
    }
    autoHideDuration={6000}
    onClose={[Function]}
    open={false}
  />
</Fragment>
`;
